File: app/src/main/java/com/ravimishra/newstar/TabAdapter.java
package com.ravimishra.newstar;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class TabAdapter extends FragmentPagerAdapter {
    private final List<Fragment> mFragment=new ArrayList<>();
    private final List<String> TitleList=new ArrayList<>();
    public TabAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return mFragment.get(i);
    }

    @Override
    public int getCount() {
        return mFragment.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return TitleList.get(position);
    }

    public void addFragment(Fragment fragment, String title){
        mFragment.add(fragment);
        TitleList.add(title);
    }
}

File: app/src/main/java/com/ravimishra/newstar/TabAdapter.kt
package com.ravimishra.newstar

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class TabAdapter(fm: FragmentManager?) : FragmentPagerAdapter(
    fm!!
) {
    private val mFragment: MutableList<Fragment> = ArrayList()
    private val TitleList: MutableList<String> = ArrayList()
    override fun getItem(i: Int): Fragment {
        return mFragment[i]
    }

    override fun getCount(): Int {
        return mFragment.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return TitleList[position]
    }

    fun addFragment(fragment: Fragment, title: String) {
        mFragment.add(fragment)
        TitleList.add(title)
    }
}